components/datetime/src/mock/mod.rs

```diff
@@ -69,6 +68,38 @@ pub fn parse_gregorian_from_str(input: &str) -> Result<DateTime<Gregorian>, Date
     Ok(datetime)
 }
 
+/// Parse a [`DateTime`] and [`MockTimeZone`] from a string.
+///
+/// This utility is for easily creating dates, not a complete robust solution. The
+/// string must take a specific form of the ISO 8601 format:
+/// `YYYY-MM-DDThh:mm:ssZ`,
+/// `YYYY-MM-DDThh:mm:ss±hh`,
+/// `YYYY-MM-DDThh:mm:ss±hhmm`,
+/// `YYYY-MM-DDThh:mm:ss±hh:mm`,
+///
+/// # Examples
+///
+/// ```
+/// use icu::datetime::mock;
+///
+/// let (date, zone) = mock::parse_zoned_gregorian_from_str("2020-10-14T13:21:00+05:30")
+///     .expect("Failed to parse a zoned datetime.");
+/// ```
+pub fn parse_zoned_gregorian_from_str(
```

Comment:
Question: Are there plans to move this into `DateTime<Iso>::parse_with_zone("...")` or similar?

Reply:
So I'm not sure about this: This presupposes a particular format, and I'm not sure if the current parser is that _good_.

Also DateTime doesn't know what a Zone is.

That said, I do plan to add a Date ctor that takes in day/month code/year/era code and something similar for DateTime

Reply:
The "particular format" should be a string conforming to [IETF Sedate](https://datatracker.ietf.org/wg/sedate/about/).

Maybe in 1.x we should make a proper well-designed parser for these strings. In the mean time, this function is poorly named and is still very much in a "mock" state. I don't have a proposal for how to fix this but I am not super excited to have this function in the critical path of using DTF in ICU4X 1.0

Reply:
https://github.com/unicode-org/icu4x/issues/2241

Reply:
@sffc It's not in the critical path; there are better ways to construct DateTimes from parts: Both `DateTime::new_gregorian_datetime()` and the function I plan to add.

It's just _convenient_ in tests.